clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -71,6 +72,7 @@ public class MockProducer<K, V> implements Producer<K, V> {
     private boolean producerFenced;
     private boolean sentOffsets;
     private long commitCount = 0L;
+    final List<KafkaMetric> addedMetrics = new ArrayList<>();
```

Comment:
Rather `default` should we mark this as private and expose a getter which should return unmodifiable copy of `addedMetrics`? Exposing getter will be as what other members in the class currently exposed as, unmodifiable copy will limit the scope of error in downstream tests by mistakenly modifying the list.

Also shall we have lazy initializtion for the member?

Same as above for other files.